client/client_debug.go

```diff
@@ -108,49 +85,79 @@ func (cli *VanClient) SkupperDump(ctx context.Context, tarName string, version s
-			writeTar("oc-versions.txt", ocv, time.Now(), tw)
+			writeTar("/skupper-info/oc-versions.yaml", ocv, time.Now(), tw)
 		}
 	}
 
-	var cversions []string
-	vir, err := cli.RouterInspect(context.Background())
+	_, err = runCommand("skupper", "version", "manifest")
 	if err == nil {
-		cversions = append(cversions, fmt.Sprintf("%-30s %s", "client version", version))
-		cversions = append(cversions, fmt.Sprintf("%-30s %s", "transport version", vir.TransportVersion))
-		cversions = append(cversions, fmt.Sprintf("%-30s %s\n", "controller version", vir.ControllerVersion))
-		writeTar("skupper-versions.txt", []byte(strings.Join(cversions, "\n")), time.Now(), tw)
+		manifest, err := os.ReadFile("./manifest.json")
+		if err == nil {
+			writeTar("/skupper-info/manifest.json", manifest, time.Now(), tw)
+			os.Remove("./manifest.json")
+		}
 	}
 
 	for i := range deployments {
-		err := cli.writeDeployment(deployments[i], tw)
+		deployment, err := cli.KubeClient.AppsV1().Deployments(cli.Namespace).Get(context.TODO(), deployments[i], metav1.GetOptions{})
+		if err != nil {
+			continue
+		}
+		err = writeObject(deployment, "/deployments/"+deployment.Name, ".yaml", tw)
 		if err != nil {
 			return dumpFile, err
 		}
 
-		component := kube.GetDeploymentLabel(deployments[i], "skupper.io/component", cli.Namespace, cli.KubeClient)
+		component, ok := deployment.Spec.Template.Labels["skupper.io/component"]
+		if !ok {
+			continue
+		}
 
-		podList, err := kube.GetPods("skupper.io/component="+component, cli.Namespace, cli.KubeClient)
-		if errors.IsNotFound(err) {
+		podList, err := cli.KubeClient.CoreV1().Pods(cli.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "skupper.io/component=" + component})
+		if err != nil {
 			continue
-		} else if err != nil {
-			return dumpFile, err
 		}
-		for _, pod := range podList {
+
+		for _, pod := range podList.Items {
+			pod, err := cli.KubeClient.CoreV1().Pods(cli.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				continue
+			} else {
+				err := writeObject(pod, "/pods/"+pod.Name+"/pod", ".yaml", tw)
+				if err != nil {
+					return dumpFile, err
+				}
+			}
+			top, err := runCommand("kubectl", "top", "pod", pod.Name, "--kubeconfig="+kubeConfigPath, "--context="+kubeConfigContext)
+			if err == nil {
+				writeTar("/pods/"+pod.Name+"/top-pod.txt", top, time.Now(), tw)
+			}
+
 			for container := range pod.Spec.Containers {
 				if pod.Spec.Containers[container].Name == "router" {
 					// while we are here collect qdstats, logs will show these operations
 					for x := range qdstatFlags {
 						qdr, err := kube.ExecCommandInContainer([]string{"skstat", qdstatFlags[x]}, pod.Name, "router", cli.Namespace, cli.KubeClient, cli.RestConfig)
 						if err == nil {
-							writeTar(pod.Name+"-skstat"+qdstatFlags[x]+".txt", qdr.Bytes(), time.Now(), tw)
+							writeTar("/pods/"+pod.Name+"/skstat/skstat-"+qdstatFlags[x]+".txt", qdr.Bytes(), time.Now(), tw)
```

Comment:
minor: the resultant files are `skstat--a.txt`, `skstat--c.txt`, etc; could it be possible to remove the extra dash after `skstat`?